### PR TITLE
Add tooltip on hover of color and typography presets

### LINKED ISSUE #61743: Add tooltip to hover for color presets
### Description

When hovering the color presets in the site editor, I expect to see a tooltip to show that tells me the title of the preset (presumably coming from the title of the style variation it's a part of).

### Step-by-step reproduction instructions

1. Go to site editor > global styles > colors > palette
2. Hover over a preset
3. See nothing

### Screenshots, screen recording, code snippet

The video below shows hover over theme colors in the palette section, which reveals tool tip. When hovering over the presets, no tool tip is revealed.

https://github.com/WordPress/gutenberg/assets/486261/f43e057c-a5f1-4846-81f9-bb94feb06608

### Environment info

WordPress 6.5.3
Gutenberg 18.4.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by amitraj2203**
## What?
Fix: #61743 

## Why?
To improve the user experience by displaying the title of each color preset when hovering over them in the site editor's global styles palette just like it happens with the theme colors.

## How?
This PR adds tooltip to the color presets in the site editor's global styles palette to display the title of each preset when hovered over.

## Testing Instructions
1. Go to site editor > global styles > colors > palette
2. Hover over a preset

## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/77401999/dfac9baf-1670-4dde-946d-ac61c4bf67c2

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/variations/variations-color.js [hunk 1]
```diff
@@ -24,9 +24,13 @@ export default function ColorVariations( { title, gap = 2 } ) {
 			{ title && <Subtitle level={ 3 }>{ title }</Subtitle> }
 			<VStack spacing={ gap }>
 				{ colorVariations.map( ( variation, index ) => (
-					<Variation key={ index } variation={ variation } isPill>
-						{ () => <StylesPreviewColors /> }
-					</Variation>
+					<Tooltip key={ index } text={ variation?.title }>
+						<div>
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/variations/variations-color.js]: Do we need this extra div?

**Comment by scruffian** [comment]
It would be good to do this for typography variations too, though happy for that to be a separate PR

File: packages/edit-site/src/components/global-styles/variations/variations-color.js (hunk 1 above)
Comment (amitraj2203) [reply packages/edit-site/src/components/global-styles/variations/variations-color.js]: Hi @scruffian, thanks for your review.

I tried it without the div, but it seems that the Tooltip doesn't seem to work then.

**Comment by amitraj2203** [comment]
> It would be good to do this for typography variations too, though happy for that to be a separate PR

Thanks for suggestion. Addressed this in a separate [PR](https://github.com/WordPress/gutenberg/pull/62678).

**Comment by bgardner** [comment]
Confirming this works as expected and is a nice touch!

File: packages/edit-site/src/components/global-styles/variations/variations-color.js [hunk 2]
```diff
@@ -27,14 +28,17 @@ export default function ColorVariations( { title, gap = 2 } ) {
...
-					<Variation
-						key={ index }
-						variation={ variation }
-						isPill
-						property="color"
-					>
-						{ () => <StylesPreviewColors /> }
-					</Variation>
+					<Tooltip key={ index } text={ variation?.title }>
+						<div>
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/variations/variations-color.js]:
```suggestion
						{ /* This div is needed for Tooltips to work */ }
						<div>
```

**Review by scruffian — APPROVED** [review]
It's a shame that we need this extra div but I can't find a way round it.

**Comment by carolinan** [comment]
Can you please include testing instructions and post the test results for users of screen readers and keyboard users.

**Comment by carolinan** [comment]
Also does this affect the color palette presets in every part of the UI; or only in the Styles sidebar as in the clip?

**Comment by amitraj2203** [comment]
Hi @scruffian & @carolinan, because of the extra div the focus is not working properly on the palletes. It requires two tab button clicks to focus the palette.
Can we move the Tooltip inside `<Variation>` that way it doesn't requies extra div and also with this change the tooltip will work on all the three - Styles, palettes and Typography. We can remove it specifically from Styles but still wants your confirmation.

This is how it's working now:

https://github.com/WordPress/gutenberg/assets/77401999/6966d90d-7a7b-4d2a-a2ae-31f7228266c5


And if we move the Tooltip inside `<Variation>` then this is how it will work:

https://github.com/WordPress/gutenberg/assets/77401999/61c9b233-976f-470c-9ebb-638f4a02b6ab

**Comment by scruffian** [comment]
> Can we move the Tooltip inside <Variation> that way it doesn't requies extra div and also with this change the tooltip will work on all the three - Styles, palettes and Typography. We can remove it specifically from Styles but still wants your confirmation

I had considered this too, I think we should try it.

**Comment by amitraj2203** [comment]
> I had considered this too, I think we should try it.

Updated it. Here's a demo,

https://github.com/WordPress/gutenberg/assets/77401999/37d0fd17-a496-4ca4-9a0e-e98a5aff6e33

**Comment by amitraj2203** [comment]
@scruffian Is anything else required from my side?


Thanks!

**Review by scruffian — APPROVED** [review]
Great, thanks for iterating on this!

**Comment by scruffian** [comment]
The PHP tests seem stuck on this PR. Since the PR doesn't touch and PHP I'm going to merge it.

**Pull request merged**